src/main/java/org/spongepowered/api/conversation/Question.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.conversation;
+
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.command.args.CommandElement;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.text.TextRepresentable;
+import org.spongepowered.api.util.ResettableBuilder;
+
+/**
+ * A {@link Conversation} contains one of these which point to each other,
+ * allowing for handling, and identification.
+ */
+public interface Question extends TextRepresentable {
+
+    /**
+     * Creates a new {@link Question.Builder} to build a {@link Question}.
+     *
+     * @return The new builder
+     */
+    static Question.Builder builder() {
+        return Sponge.getRegistry().createBuilder(Question.Builder.class);
+    }
+
+    /**
+     * Gets the id of this question. Should be different from all
+     * other questions within its {@link Conversation};
+     *
+     * @return The id
+     */
+    String getId();
+
+    /**
+     * Gets the handler which handles a {@link Conversant}'s answer
+     * to this question.
+     *
+     * @return The answer handler for this question
+     */
+    AnswerHandler getHandler();
+
+    /**
+     * Gets the prompt for this question.
+     *
+     * @return The prompt
+     */
+    Text getPrompt();
```

Comment:
It'd be neat to be able to make use of the current context within the prompt messages. For example, question #1 might ask for the conversant's name, then question #2 could be like: "hi <name>, how old are you?".

Maybe prompts could be a Function<Context,Text>-like interface rather than preset Texts?